### DISCUSSION TITLE
code/file organization

### DISCUSSION THREAD

**Discussion by ogrotten** [replit/kaboom#684] [Q&A]
What is the "proper" way to extract various methods to other files?

Take `loadSpriteAtlas` for example. If I put it in an external file like 

```
export const ruins = loadSpriteAtlas('/assets/Ruins/Ruins.png', {. . .})
```

it doesn't work as expected if I then import and run `ruins()`

Similarly, 

```
scene('start', setup)
```

does not work if the function `setup` is defined elsewhere, imported, and passed in like above.

So what is the proper way to lighten the load on the overburdened `main.js` file?

**Comment by lajbel** [comment]
`kaboom()` exports all kaboom's functions, so, you need "import" it after the kaboom run. But because you only can import things in top of the file, you should import a function, so you can use

`loader.js`
```js
export const function loadRuin() {
    return loadSpriteAtlas('/assets/Ruins/Ruins.png', {. . .});
}
```

`main.js`
```js
import kaboom from "kaboom";
import { loadRuin } from "./loeader.js";

kaboom();

// load all after kaboom
loadRuin();
```
